src/main/frontend/handler/editor.cljs

```diff
@@ -834,28 +834,38 @@
                        ;; it's possible to find a block not belong to the same page of current editing block
                        sibling-block (util/get-prev-block-non-collapsed-non-embed block-parent)
                        delete_prev? (and (not (block-has-no-ref? (:db/id block)))
-                                         (block-has-no-ref? [:block/uuid (uuid (dom/attr sibling-block "blockid"))]))
+                                         (block-has-no-ref? [:block/uuid (uuid (dom/attr sibling-block "blockid"))])
+                                         (not (string/blank? value))) ; it doesn't make sense to preserve the block ref for totally different block
```

Comment:
I have a different opinion.

Here is my two considerations:
1. change the relationship of blocks and not modify the ID and UUID as much as possible.
2. make sure not to disrupt any references if possible.

Moreover, whether the block is empty or not is not taken into consideration since it only pertains to a change in content.

By keeping the reference, users can easily locate and access the referenced blocks by clicking the open-block-ref-link and perform any necessary operations.

Reply:
I'll take some time to prepare an ADR on broken refs handling first, let's discuss about those things later 😄